qa/integration/specs/settings_spec.rb

```diff
@@ -0,0 +1,139 @@
+    end
+    expect(File.exists?("#{temp_dir}/logstash-plain.log")).to be true
+  end
+  
+  it "should read config from the specified dir in logstash.yml" do
+    change_setting("path.config", temp_dir)
+    test_config_path = File.join(temp_dir, "test.config")
+    IO.write(test_config_path, tcp_config)
+    expect(File.exists?(test_config_path)).to be true
+    @logstash_service.spawn_logstash
+    # check LS is up and running with new data path
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(is_port_open?(test_port)).to be true
+    end
+  end
+  
+  it "should exit when config test_and_exit is set" do
+    s = {}
+    s["path.config"] = temp_dir
+    s["config.test_and_exit"] = true
+    s["path.logs"] = temp_dir
+    overwrite_settings(s)
+    test_config_path = File.join(temp_dir, "test.config")
+    IO.write(test_config_path, "#{tcp_config}")
+    expect(File.exists?(test_config_path)).to be true
+    @logstash_service.spawn_logstash
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(@logstash_service.exited?).to be true
+    end
+    expect(@logstash_service.exit_code).to eq(0)
+    
+    # now with bad config
+    IO.write(test_config_path, "#{tcp_config} filters {} ")
+    expect(File.exists?(test_config_path)).to be true
+    @logstash_service.spawn_logstash
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(@logstash_service.exited?).to be true
+    end
+    expect(@logstash_service.exit_code).to eq(1)
+  end
+  
+  it "change pipeline settings" do
+    s = {}
+    workers = 31
+    batch_size = 1250
+    s["pipeline.workers"] = workers
+    s["pipeline.batch.size"] = batch_size
+    overwrite_settings(s)
+    @logstash_service.spawn_logstash("-e", tcp_config)
+    @logstash_service.wait_for_logstash
+    # check LS is up and running with new data path
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(is_port_open?(test_port)).to be true
+    end
+    
+    # now check monitoring API to validate
+    node_info = @logstash_service.monitoring_api.node_info
+    expect(node_info["pipeline"]["workers"]).to eq(workers)
+    expect(node_info["pipeline"]["batch_size"]).to eq(batch_size)
+  end
+  
+  it "start on a different HTTP port" do
+    # default in 9600
+    http_port = random_port
+    change_setting("http.port", http_port)
+    @logstash_service.spawn_logstash("-e", tcp_config)
+    
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(is_port_open?(http_port)).to be true
+    end
+    # check LS is up and running with new data path
+    Stud.try(retry_attempts.times, RSpec::Expectations::ExpectationNotMetError) do
+      expect(is_port_open?(test_port)).to be true
+    end
+    
+    resp = Manticore.get("http://localhost:#{http_port}/_node").body
+    node_info = JSON.parse(resp)
```

Comment:
https://github.com/elastic/logstash/pull/6031/files#diff-253716b090f409a88378bf42eb762a96R25 maybe we can add a port option?